arch/arm/soc/st_stm32/stm32l4/soc_gpio.c

```diff
@@ -70,86 +49,28 @@ struct stm32l4x_gpio {
 	u32_t ascr; /* Only present on STM32L4x1, STM32L4x5, STM32L4x6 */
 };
 
-/**
- * @brief map pin function to MODE register value
- */
-static u32_t func_to_mode(int conf, unsigned int afnum)
-{
-	/* If an alternate function is specified */
-	if (afnum) {
-		return STM32L4X_MODER_ALT_MODE;
-	}
-
-	switch (conf) {
-	case STM32L4X_PIN_CONFIG_BIAS_HIGH_IMPEDANCE:
-	case STM32L4X_PIN_CONFIG_BIAS_PULL_UP:
-	case STM32L4X_PIN_CONFIG_BIAS_PULL_DOWN:
-		return STM32L4X_MODER_INPUT_MODE;
-	case STM32L4X_PIN_CONFIG_ANALOG:
-		return STM32L4X_MODER_ANALOG_MODE;
-	default:
-		return STM32L4X_MODER_OUTPUT_MODE;
-	}
-
-	return STM32L4X_MODER_INPUT_MODE;
-}
-
-static u32_t func_to_otype(int conf)
-{
-	switch (conf) {
-	case STM32L4X_PIN_CONFIG_OPEN_DRAIN:
-	case STM32L4X_PIN_CONFIG_OPEN_DRAIN_PULL_UP:
-	case STM32L4X_PIN_CONFIG_OPEN_DRAIN_PULL_DOWN:
-		return STM32L4X_OTYPER_OPEN_DRAIN;
-	default:
-		return STM32L4X_OTYPER_PUSH_PULL;
-	}
-
-	return STM32L4X_OTYPER_PUSH_PULL;
-}
-
-static u32_t func_to_pupd(int conf)
-{
-	switch (conf) {
-	case STM32L4X_PIN_CONFIG_ANALOG:
-	case STM32L4X_PIN_CONFIG_BIAS_HIGH_IMPEDANCE:
-	case STM32L4X_PIN_CONFIG_PUSH_PULL:
-	case STM32L4X_PIN_CONFIG_OPEN_DRAIN:
-		return STM32L4X_PUPDR_NO_PULL;
-	case STM32L4X_PIN_CONFIG_BIAS_PULL_UP:
-	case STM32L4X_PIN_CONFIG_PUSH_PULL_PULL_UP:
-	case STM32L4X_PIN_CONFIG_OPEN_DRAIN_PULL_UP:
-		return STM32L4X_PUPDR_PULL_UP;
-	case STM32L4X_PIN_CONFIG_BIAS_PULL_DOWN:
-	case STM32L4X_PIN_CONFIG_PUSH_PULL_PULL_DOWN:
-	case STM32L4X_PIN_CONFIG_OPEN_DRAIN_PULL_DOWN:
-		return STM32L4X_PUPDR_PULL_DOWN;
-	}
-
-	return STM32L4X_PUPDR_NO_PULL;
-}
-
 int stm32_gpio_flags_to_conf(int flags, int *pincfg)
 {
 	int direction = flags & GPIO_DIR_MASK;
+	int pud = flags & GPIO_PUD_MASK;
 
 	if (!pincfg) {
 		return -EINVAL;
 	}
 
 	if (direction == GPIO_DIR_OUT) {
-		*pincfg = STM32L4X_PIN_CONFIG_PUSH_PULL;
+		*pincfg = STM32_MODER_OUTPUT_MODE;
 	} else {
-		int pud = flags & GPIO_PUD_MASK;
-
 		/* pull-{up,down} maybe? */
+		*pincfg = STM32_MODER_INPUT_MODE;
+
 		if (pud == GPIO_PUD_PULL_UP) {
-			*pincfg = STM32L4X_PIN_CONFIG_BIAS_PULL_UP;
+			*pincfg = *pincfg | STM32_PUPDR_PULL_UP;
```

Comment:
What about output pull up or down? It seems like this only allows pull in the input direction.